pkg/detector/ospkg/wolfi/wolfi_test.go

```diff
@@ -0,0 +1,241 @@
+						URL:  "https://packages.wolfi.dev/os/security.json/",
+					},
+				},
+				{
+					PkgName:          "ansible",
+					VulnerabilityID:  "CVE-2021-20191",
+					InstalledVersion: "2.6.4",
+					FixedVersion:     "",
+					Layer: ftypes.Layer{
+						DiffID: "sha256:932da51564135c98a49a34a193d6cd363d8fa4184d957fde16c9d8527b3f3b02",
+					},
+					DataSource: &dbTypes.DataSource{
+						ID:   vulnerability.Wolfi,
+						Name: "Wolfi Secdb",
+						URL:  "https://packages.wolfi.dev/os/security.json/",
+					},
+				},
+			},
+		},
+		{
+			name:     "contain rc",
+			fixtures: []string{"testdata/fixtures/wolfi.yaml", "testdata/fixtures/data-source.yaml"},
+			args: args{
+				pkgs: []ftypes.Package{
+					{
+						Name:       "jq",
+						Version:    "1.6-r0",
+						SrcName:    "jq",
+						SrcVersion: "1.6-r0",
+					},
+				},
+			},
+			want: []types.DetectedVulnerability{
+				{
+					PkgName:          "jq",
+					VulnerabilityID:  "CVE-2020-1234",
+					InstalledVersion: "1.6-r0",
+					FixedVersion:     "1.6-r1",
+					DataSource: &dbTypes.DataSource{
+						ID:   vulnerability.Wolfi,
+						Name: "Wolfi Secdb",
+						URL:  "https://packages.wolfi.dev/os/security.json/",
+					},
+				},
+			},
+		},
+		{
+			name:     "contain pre",
+			fixtures: []string{"testdata/fixtures/wolfi.yaml", "testdata/fixtures/data-source.yaml"},
+			args: args{
+				pkgs: []ftypes.Package{
+					{
+						Name:       "test",
+						Version:    "0.1.0_alpha",
+						SrcName:    "test-src",
+						SrcVersion: "0.1.0_alpha",
+						Layer: ftypes.Layer{
+							DiffID: "sha256:932da51564135c98a49a34a193d6cd363d8fa4184d957fde16c9d8527b3f3b02",
+						},
+					},
+				},
+			},
+			want: []types.DetectedVulnerability{
+				{
+					VulnerabilityID:  "CVE-2030-0002",
+					PkgName:          "test",
+					InstalledVersion: "0.1.0_alpha",
+					FixedVersion:     "0.1.0_alpha2",
+					Layer: ftypes.Layer{
+						DiffID: "sha256:932da51564135c98a49a34a193d6cd363d8fa4184d957fde16c9d8527b3f3b02",
+					},
+					DataSource: &dbTypes.DataSource{
+						ID:   vulnerability.Wolfi,
+						Name: "Wolfi Secdb",
+						URL:  "https://packages.wolfi.dev/os/security.json/",
+					},
+				},
+			},
+		},
+		{
+			name:     "repository is newer than OS version",
+			fixtures: []string{"testdata/fixtures/wolfi.yaml", "testdata/fixtures/data-source.yaml"},
+			args: args{
+				repo: &ftypes.Repository{
+					Family: os.Wolfi,
+				},
+				pkgs: []ftypes.Package{
+					{
+						Name:       "jq",
+						Version:    "1.6-r0",
+						SrcName:    "jq",
+						SrcVersion: "1.6-r0",
+					},
+				},
+			},
+			want: []types.DetectedVulnerability{
+				{
+					PkgName:          "jq",
+					VulnerabilityID:  "CVE-2020-1234",
+					InstalledVersion: "1.6-r0",
+					FixedVersion:     "1.6-r1",
+					DataSource: &dbTypes.DataSource{
+						ID:   vulnerability.Wolfi,
+						Name: "Wolfi Secdb",
+						URL:  "https://packages.wolfi.dev/os/security.json/",
+					},
+				},
+			},
+		},
```

Comment:
This test case is for Alpine Linux. We can remove.

Reply:
Removed in a9f359ff